### ISSUE TITLE
Can you write code that trains bert with MLM and the next sentence at the same time?

### ISSUE THREAD

**Issue by xiao12mm** [huggingface/transformers#23177]
### Feature request

Or is there a training code already in place?

### Motivation

I've learned that training with both is better than training with mlm alone, by which I mean the generated vector features

### Your contribution

no yet

**Comment by AKBakshay** [comment]
Hi @xiao12mm, I belive the NSP loss isn't available for model training directly in huggingface. This [thread](https://discuss.huggingface.co/t/continual-pre-training-from-an-initial-checkpoint-with-mlm-and-nsp/6869) provide a script for NSP training.

About this - "training with both is better than training with mlm alone", there has been specific research done to verify the claim. One of the conclusions from "RoBERTa: A Robustly Optimized BERT Pretraining Approach" paper is that - "removing the NSP loss matches or slightly improves downstream task performance".
